src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import TestUtils from 'react-dom/test-utils';
+import Decimal from './decimal.component';
+import Textbox from '../textbox/textbox.component';
+
+function render(props) {
+  const onChange = props.onChange ? props.onChange : () => true;
+
+  return shallow(
+    <Decimal
+      onChange={ onChange }
+      value={ props.value }
+      precision={ props.precision }
+    />
+  );
+}
+
+function renderIntoDocument(props) {
+  return TestUtils.renderIntoDocument(
+    <Decimal value={ props.value } precision={ props.precision } />
+  );
+}
+
+describe('Decimal', () => {
+  describe('Input validation', () => {
+    it('renders the correct value', () => {
+      const input = render({ value: '9.87' });
+      expect(input.find(Textbox).prop('value')).toEqual('9.87');
+    });
+
+    it('renders a value from props exactly even when invalid', () => {
+      const input = render({ value: '12abc.85' });
+      expect(input.find(Textbox).prop('value')).toEqual('12abc.85');
+    });
+
+    it('updates an external state onChange', () => {
+      const onChange = jest.fn();
+      const input = render({ onChange });
+      input.instance().onChange({ target: { value: '14.79' } });
+      expect(onChange).toHaveBeenCalledWith({ target: { value: '14.79' } });
+    });
+
+    it('does not allow the user to enter letters or special characters', () => {
+      const input = render({ value: '12.34' });
+      input.instance().onChange({ target: { value: '1hello$1.27' } });
```

Comment:
Do we have any existing helpers or documentation for what characters we should be testing?

Reply:
I don't believe so, feel free to create some. 

Otherwise `!@£$%^&*()#_+=}{|":;'\<>?`
`qwertyuiopasdfghjklzxcvbnm` (with toUpperCase) should be fine. 

It's just that, if we're going to test for special chars, there's no point only checking for one.